stdlib/public/core/HashedCollections.swift.gyb

```diff
@@ -268,20 +268,15 @@ internal var _hashContainerDefaultMaxLoadFactorInverse: Double {
 internal func _stdlib_NSObject_isEqual(_ lhs: AnyObject, _ rhs: AnyObject) -> Bool
 #endif
 
-//===--- Hacks and workarounds --------------------------------------------===//
-
-// FIXME(ABI)#158 (Performance)
-// There used to be a now-fixed optimizer bug that prevented using
-// UnsafePointer<Unmanaged<AnyObject>> in Dictionary implementation.
-// rdar://problem/16861508
-/// Like `UnsafeMutablePointer<Unmanaged<AnyObject>>`, or `id
-/// __unsafe_unretained *` in Objective-C ARC.
+/// A convenience for using `UnsafeMutablePointer<Unmanaged<AnyObject>>`
+/// (`id __unsafe_unretained *` in Objective-C ARC).
 internal struct _UnmanagedAnyObjectArray {
   /// Underlying pointer.
-  internal var value: UnsafeMutableRawPointer
+  internal var value: UnsafeMutablePointer<Unmanaged<AnyObject>>
 
   internal init(_ up: UnsafeMutablePointer<AnyObject>) {
-    self.value = UnsafeMutableRawPointer(up)
+    self.value = UnsafeMutableRawPointer(up).assumingMemoryBound(
+        to: Unmanaged<AnyObject>.self)
```

Comment:
The other hope is that someone could use the info to write a sanitizer-like dynamic analysis some day.